dev/com.ibm.ws.jsp/src/com/ibm/ws/jsp/translator/visitor/JspVisitor.java

```diff
@@ -324,9 +325,15 @@ else if (jspElementType.equals(Constants.JSP_SETPROPERTY_TYPE)) {
                     visitJspSetPropertyEnd(jspElement);
                 }
                 else if (jspElementType.equals(Constants.JSP_PLUGIN_TYPE)) {
-                    visitJspPluginStart(jspElement);
-                    processChildren(jspElement);
-                    visitJspPluginEnd(jspElement);
+                    if(PagesVersionHandler.isPages30OrLowerLoaded()){
+                        visitJspPluginStart(jspElement);
+                        processChildren(jspElement);
+                        visitJspPluginEnd(jspElement);
+                    } else {
+                        if(com.ibm.ejs.ras.TraceComponent.isAnyTracingEnabled()&&logger.isLoggable(Level.FINEST)){
+                            logger.logp(Level.FINEST, CLASS_NAME, "processJspElement","Skipping jsp:plugin.");
```

Comment:
Can you expand the message to say why the skip is happening?

Reply:
`jsp:plugin `is now a no-op as current browsers don't support the generated html tags `https://github.com/jakartaee/pages/issues/226`

Reply:
Updated to: `Skipping jsp:plugin element as it is a no-op for Pages 3.1`